pkg/reconciler/buildrun/buildrun.go

```diff
@@ -92,16 +91,29 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 			}
 
 			build = &buildv1alpha1.Build{}
-			if err = resources.GetBuildObject(ctx, r.client, buildRun.Spec.BuildRef.Name, buildRun.Namespace, build); err != nil {
-				updateErr := r.updateBuildRunErrorStatus(ctx, buildRun, err.Error())
-				return reconcile.Result{}, resources.HandleError("Failed to fetch the Build instance", err, updateErr)
+			if err = resources.GetBuildObject(ctx, r.client, buildRun, build); err != nil {
+				// system call failure, reconcile again
+				return reconcile.Result{}, err
 			}
 
 			// Validate if the Build was successfully registered
-			if err := r.ValidateBuildRegistration(ctx, build, buildRun); err != nil {
+			if build.Status.Registered == "" {
+				err := fmt.Errorf("the Build is not yet validated, build: %s", build.Name)
+				// reconcile again until it gets a registration value
 				return reconcile.Result{}, err
 			}
 
+			if build.Status.Registered != corev1.ConditionTrue {
+				// stop reconciling and mark the BuildRun as Failed
+				// we only reconcile again if the status.Update call fails
+				message := fmt.Sprintf("the Build is not registered correctly, build: %s, registered status: %s, reason: %s", build.Name, build.Status.Registered, build.Status.Reason)
+				if updateErr := resources.UpdateConditionWithFalseStatus(ctx, r.client, buildRun, message, resources.ConditionBuildRegistrationFailed); updateErr != nil {
+					return reconcile.Result{}, updateErr
+				}
+
+				return reconcile.Result{}, nil
```

Comment:
It can, but is on purpose, otherwise reading this function might be confusing. While the resources.UpdateConditionWithFalseStatus() can return an error by itself. I would like to keep it this the same way.

Reply:
That's fine, both works. I am still trying to find out when the "go community" seeks for writing it as short as possible at all costs or when one prefers the more detailed code flow. :-)